zeebe/gateway/src/test/java/io/camunda/zeebe/gateway/api/decision/EvaluateDecisionStub.java

```diff
@@ -66,9 +66,9 @@ public void registerWith(final StubbedBrokerClient gateway) {
   @Override
   public BrokerResponse<DecisionEvaluationRecord> handle(
       final BrokerEvaluateDecisionRequest request) throws Exception {
-    // In practise the decision instance key is generated newly by the broker
-    final long decisionInstanceKey = DECISION_RECORD.getDecisionKey() + 1;
```

Comment:
I didn't get how this was relevant, it was test setup code that increased the key value which was then validated in the test, I would argue there is no point in testing test setup code? I thus removed the modification of the key here.

Reply:
That was also my impression a bit. We can also ask @koevskinikola if he still knows 🙂

Reply:
Found that piece was added by @korthout on main recently https://github.com/camunda/zeebe/commit/164c854c484317084ae2e801d7c9b9c4bfb99a0c let's sync on whether we wanna add it back.

Reply:
Ah, I see, good catch 👍 I think we should re-add this then, yeah.

Reply:
Would like to understand the reason for stub code that increments the key to be validated by the test still.

Reply:
Each decision evaluation receives a unique key from the broker. This key is different from the decision key, which uniquely identifies the decision to evaluate. Rather it is the unique evaluation (or instance) of that decision.

This key is responded back to the client so the user can use it to find the decision instance (the evaluated decision) in Operate. 

The test doesn't need to be this specific to pass, but it provides a more realistic test case. It's just more correct semantically.